endpoints/feature-extraction-service.yml

```diff
@@ -0,0 +1,130 @@
+        "5XX":
+          description: Server error.
+          content:
+            application/json:
+              schema:
+                $ref: "../components/Error.yaml"
+components:
+  schemas:
+    FeatureExtractionParameters:
+      required:
+        - featureType
+        - image
+        - intrinsics
+        - maxNumFeatures
+      properties:
+        featureType:
+          $ref: "#/components/schemas/FeatureType"
+        gravity:
+          $ref: "#/components/schemas/Gravity"
+        image:
+          description: Target image for feature extraction.
+          type: string
+          format: binary
+        intrinsics:
+          $ref: "../components/Intrinsics.yaml"
+        maxNumFeatures:
+          description: Maximum number of features to extract.
+          type: number
+          format: integer
+        maxSidelength:
+          description: Greater of the height or width of the image in pixels which serves as an optional scaling parameter for the map. If both maxSidelength and targetFocalLength are provided, targetFocalLength is preferred.
+          type: number
+          format: integer
+        targetFocalLength:
+          description: Desired focal length of the image which serves as an optional scaling parameter for the map. If both maxSidelength and targetFocalLength are provided, targetFocalLength is preferred.
+          type: number
+          format: float
+    FeatureExtractionResult:
+      required:
+        - featureType
+        - intrinsics
+        - numFeatures
+      properties:
+        featureDescriptors:
+          description: A 2d matrix of feature descriptors in row-major order with a total size of numFeatures * featureDimension * featureDepth bytes. Each row represents a single descriptor with featureDimension entries.
+          type: string
+          format: binary
+        featureDepth:
+          description: Feature depth.
+          enum:
+            - uint8
+            - float32
+        featureDimension:
+          description: Feature dimension.
+          type: number
+          format: integer
+        featureType:
+          $ref: "#/components/schemas/FeatureType"
+        keypointLocations:
+          description: A 2d matrix of keypoint locations in row-major order with a total size of numFeatures * 2 * 4 bytes. Each row represents the x,y coordinate of the respective descriptor in the scaled image represented as a single precision float.
+          type: string
+          format: binary
+        numFeatures:
+          description: Number of features extracted.
+          type: number
+          format: integer
+        intrinsics:
+          $ref: "../components/Intrinsics.yaml"
+    FeatureType:
+      description: Feature type to extract.
+      enum:
+        - d2net_single_scale_00_05        # the default D2-net feature descriptor with a single scale and each raw dimension clipped to a range of [0-0.5]
+    Gravity:
```

Comment:
Should `Gravity` be a component instead of being defined here?

Reply:
no, because gravity is a specific instance of orientation. orientation should be the component.